File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    count = [0] * (number + 1)  # 각 숫자의 약수의 개수를 담을 리스트 초기화
    result = 0

    for i in range(1, number+1):
        for j in range(i, number+1, i):  # i의 배수들에 대해 약수의 개수를 증가
            count[j] += 1
    
    for i in count:
        if i > limit:
            result += power
        else:
            result += i

    return result

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    count = [];
    result = 0;
    
    for (i = 0; i < number + 1; i++) { // 0부터 number까지
        count.push(0); // 0으로 초기화된 요소를 배열에 추가
    }

    for (i = 1; i <= number; i++) { // 숫자 1부터 number까지 약수 구하기
        for (j = i; j <= number; j += i) { // i부터 number까지 i씩 증가
            count[j] += 1;
        }
    }

    for (i of count) { // limit 보다 크면 power 더하기, 아니면 i 그대로 더하기
        if (i > limit) {
            result += power;
        } else {
            result += i;
        }
    }

    return result;
}